### PR TITLE
Made stop method for uploading and experiment

### PR DESCRIPTION
**Pull request by Cooksen**

### CODE REVIEW

File: decanter_ai_sdk/client.py [hunk 1]
```diff
@@ -520,6 +520,14 @@ def predict_ts(
         )
         return prediction
 
+    def stop_uploading(self, id):
```
Comment (jwolf9) [inline decanter_ai_sdk/client.py:523]: give a type hint for the return type. Maybe it should be a boolean about whether the stop was successful or not

File: decanter_ai_sdk/client.py [hunk 2]
```diff
@@ -520,6 +520,14 @@ def predict_ts(
...
+        logging.info(self.api.stop_uploading(id))
+        return
+
+    def stop_training(self, id):
+        logging.info(self.api.stop_training(id))
+        return
+
```
Comment (jwolf9) [inline decanter_ai_sdk/client.py:530]: have a stop for prediction too

**Review by jwolf9 — CHANGES_REQUESTED** [review]

File: decanter_ai_sdk/client.py (hunk 1 above)
Comment (Cooksen) [reply decanter_ai_sdk/client.py:523]: I'm thinking maybe I can use `res.ok` to detect the response of the server? and if it's ok I can just log the success message from the server, if it's not then I can log some error message ?

File: decanter_ai_sdk/client.py [hunk 3]
```diff
@@ -520,12 +520,18 @@ def predict_ts(
         )
         return prediction
 
-    def stop_uploading(self, id):
-        logging.info(self.api.stop_uploading(id))
+    def stop_uploading(self, id: str):
```
Comment (jwolf9) [inline decanter_ai_sdk/client.py:523]: maybe log the id of the task that you're stopping. Do you think that'd be good?

Comment (jwolf9) [inline decanter_ai_sdk/client.py:523]: have a type hint for the return type

**Review by jwolf9 — CHANGES_REQUESTED** [review]

File: decanter_ai_sdk/client.py [hunk 4]
```diff
@@ -520,6 +520,20 @@ def predict_ts(
         )
         return prediction
 
+    def stop_uploading(self, id: str):
```
Comment (jwolf9) [inline decanter_ai_sdk/client.py:523]: is the id logged for training and setup tasks?

Comment (Cooksen) [reply decanter_ai_sdk/client.py:523]: yes

File: decanter_ai_sdk/client.py (hunk 3 above)
Comment (Cooksen) [reply decanter_ai_sdk/client.py:523]: Yes I think that wiil be more clear!

File: decanter_ai_sdk/web_api/decanter_api.py [hunk 5]
```diff
@@ -147,3 +147,21 @@ def get_model_list(self, experiment_id, query):  # pragma: no cover
...
+    def stop_uploading(self, id):  # pragma: no cover
+        res = requests.post(
+            f"{self.url}table/stop",
+            headers=self.auth_headers,
+            verify=False,
+            data={"table_id": id, "project_id": self.project_id},
+        )
+        return res.ok
+
+    def stop_training(self, id):  # pragma: no cover
```
Comment (jwolf9) [inline decanter_ai_sdk/web_api/decanter_api.py:160]: the return type here is mismatched with the client

**Review by jwolf9 — CHANGES_REQUESTED** [review]

**Review by jwolf9 — APPROVED** [review]

**Pull request merged**